server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -146,7 +147,7 @@ public class Coordinator extends AbstractLifecycleComponent implements ClusterSt
     private final MasterServiceTaskQueue<NodeLeftExecutor.Task> nodeLeftQueue;
     private final Supplier<CoordinationState.PersistedState> persistedStateSupplier;
     private final NoMasterBlockService noMasterBlockService;
-    final Object mutex = new Object(); // package-private to allow tests to call methods that assert that the mutex is held
+    public final Object mutex = new Object(); // public to allow tests to call methods that assert that the mutex is held
```

Comment:
I don't think we should make these visibility changes to the production code, if it is only for the test PR. We could for example just add some helper e.g. 
```java
       public void becomeCandidate(ClusterNode node, String reason) {
            synchronized (node.coordinator.mutex) {
                node.coordinator.becomeCandidate(reason);
            }
        }
```
in `AbstractCoordinatorTestCase.Cluster`.